driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -68,6 +68,9 @@
     private final AtomicBoolean isDone = new AtomicBoolean();
     private AtomicInteger executionCount = new AtomicInteger();
 
+    // Timestamp common to all Speculative Executions of a same request
+    private long commonTimestamp = 0;
```

Comment:
I think this field could be made volatile; it is written by the first spec exec and read by all subsequent ones, so in theory it is safe even if it is not volatile, but just to be on the safe side (e.g. if one day someone adds another write to this value).